tests/core/io/test_resource.h

```diff
@@ -200,19 +200,27 @@ class DuplicateGuineaPigData : public Object {
 			} else if (p_a.get_type() == Variant::ARRAY) {
 				const Array &arr_a = p_a;
 				const Array &arr_b = p_b;
-				CHECK(!arr_a.is_same_instance(arr_b));
-				CHECK(arr_a.size() == arr_b.size());
-				for (int i = 0; i < arr_a.size(); i++) {
-					_verify_deep_copied_variants(arr_a[i], arr_b[i]);
+				if (p_property_usage & PROPERTY_USAGE_NEVER_DUPLICATE) {
+					CHECK(arr_a.is_same_instance(arr_b));
+				} else {
+					CHECK(!arr_a.is_same_instance(arr_b));
+					CHECK(arr_a.size() == arr_b.size());
+					for (int i = 0; i < arr_a.size(); i++) {
+						_verify_deep_copied_variants(arr_a[i], arr_b[i]);
+					}
```

Comment:
there should be similar check and behaviour for PROPERTY_USAGE_ALWAYS_DUPLICATE?

Reply:
I think `PROPERTY_USAGE_ALWAYS_DUPLICATE` only makes sense for Resources. Unless there is an exception that skips some Arrays/Dictionaries I'm not aware of. For Resources it's the `RESOURCE_DEEP_DUPLICATE_INTERNAL` flag, which makes difference between internal and external resources.